pkgs/development/libraries/qt-6/modules/qtbase.nix

```diff
@@ -0,0 +1,555 @@
+
+  patch-cmake-files-sh = ../patch-cmake-files.sh;
+  patch-cmake-files-regex-diff = ../patch-cmake-files.regex.diff;
+
+drv1 =
+stdenv.mkDerivation rec {
+
+  # when splitting, disable all phases after buildPhase
+  # https://github.com/NixOS/nixpkgs/blob/master/pkgs/stdenv/generic/setup.sh#L1305
+  dontCheck = splitBuildInstallEnabled;
+  dontInstall = splitBuildInstallEnabled;
+  dontFixup = splitBuildInstallEnabled;
+  dontInstallCheck = splitBuildInstallEnabled;
+  dontDist = splitBuildInstallEnabled;
+  postPhases = if splitBuildInstallEnabled then "splitBuildInstallLoad splitBuildInstallExport" else "";
+
+  # TODO better? how to load "hooks" for mkDerivation?
+  splitBuildInstallLoad = ''
+    . ${split-build-install-export-sh}
+  '';
+
+  pname = "qtbase";
+
+  inherit qtCompatVersion src version;
+
+  debug = debugSymbols;
+
+  propagatedBuildInputs = [
+    libxml2
+    libxslt
+    openssl
+    sqlite
+    sqlite.out
+    sqlite.dev
+    zlib
+    unixODBC
+
+    # Text rendering
+    harfbuzz
+    icu
+
+    # Image formats
+    libjpeg
+    libpng
+    pcre2
+    pcre # for glib-2.0
+    libproxy
+    libproxy.dev # pulls kerberos
+    xlibsWrapper
+    xlibs.libXdmcp
+    xlibs.libXtst
+    xlibs.xcbutilcursor
+    zstd
+    double-conversion
+    util-linux # mount for gio-2.0
+    #journalctl
+    systemd # pulls kerberos
+    libb2
+    md4c
+    mtdev
+    lksctp-tools
+    libselinux
+    libsepol
+
+    lttng-ust # linux trace
+
+    # TODO enable vulkan/openvg only when openGL is available
+    vulkan-headers
+    vulkan-loader
+
+    libthai # for pango
+    libdrm
+    libdatrie # for libthai
+    epoxy # for gdk-3.0
+    #valgrind # for libdrm. TODO add? or is it bloat?
+    #tslib # touchscreen
+  ] ++ (with unixODBCDrivers; [
+    psql
+    sqlite
+    mariadb
+  ]) ++ (
+    if stdenv.isDarwin then [
+      # TODO: move to buildInputs, this should not be propagated.
+      AGL
+      AppKit
+      ApplicationServices
+      Carbon
+      Cocoa
+      CoreAudio
+      CoreBluetooth
+      CoreLocation
+      CoreServices
+      DiskArbitration
+      Foundation
+      OpenGL
+      libobjc
+      libiconv
+      MetalKit
+      IOKit
+    ] else [
+      dbus
+      glib
+      udev
+
+      # Text rendering
+      fontconfig
+      freetype
+
+      # X11 libs
+      libX11
+      libXcomposite
+      libXext
+      libXi
+      libXrender
+      libxcb
+      libxkbcommon
+      xcbutil
+      xcbutilimage
+      xcbutilkeysyms
+      xcbutilrenderutil
+      xcbutilwm
+    ] ++ lib.optional libGLSupported libGL
+  );
+
+  buildInputs = [ python3 at-spi2-core ]
+    ++ lib.optionals (!stdenv.isDarwin)
+    (
+      [ libinput ]
+        ++ lib.optional withGtk3 gtk3
+    )
+    ++ lib.optional developerBuild gdb
+    ++ lib.optional (cups != null) cups
+    ++ lib.optional (libmysqlclient != null) libmysqlclient
+    ++ lib.optional (postgresql != null) postgresql
```

Comment:
Overwriting things with null in inputs is a relict from before callPackage. Can we please use options for such things?